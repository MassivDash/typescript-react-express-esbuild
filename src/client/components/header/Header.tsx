import React from 'react';
import { Navbar } from '@components';

const Header: React.FunctionComponent = () => {
  return (
    <header>
      <Navbar />
    </header>
  );
};

export default Header;
