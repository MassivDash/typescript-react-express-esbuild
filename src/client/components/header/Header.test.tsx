import { render } from '@testing-library/react';
import * as React from 'react';
import Header from './Header';
import { BrowserRouter } from 'react-router-dom';

describe('Header', () => {
  jest.mock('react-router-dom', () => ({
    useLocation: jest.fn().mockReturnValue({
      pathname: '/',
      search: '',
      hash: '',
      state: null,
      key: '5nvxpbdafa',
    }),
  }));

  it('should render', () => {
    const { asFragment, queryByText } = render(
      <BrowserRouter>
        <Header />
      </BrowserRouter>,
    );
    expect(asFragment()).toMatchSnapshot();
    expect(queryByText('Home')).toBeTruthy();
  });
});
