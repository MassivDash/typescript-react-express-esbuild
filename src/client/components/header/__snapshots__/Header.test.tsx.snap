// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render 1`] = `
<DocumentFragment>
  <header>
    <nav
      class="bg-gray-800 w-full h-14 flex justify-start items-center p-2"
    >
      <div
        class="ml-3 bg-gray-800 flex text-sm rounded-full focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-offset-gray-800 focus:ring-white"
      >
        <img
          alt="Spaceghost"
          class="h-8 w-8 rounded-full"
          src="https://blog.spaceout.pl/content/images/2020/03/spaceghost-1.jpg"
        />
      </div>
      <ul
        class="flex ml-2  w-56 justify-evenly"
      >
        <li
          class="text-white px-3 py-2 rounded-md text-sm font-medium hover:bg-gray-700 hover:text-white bg-gray-900"
        >
          <a
            href="/"
          >
            Home
          </a>
        </li>
        <li
          class="text-white px-3 py-2 rounded-md text-sm font-medium hover:bg-gray-700 hover:text-white false"
        >
          <a
            href="/about"
          >
            About
          </a>
        </li>
        <li
          class="text-white px-3 py-2 rounded-md text-sm font-medium hover:bg-gray-700 hover:text-white false"
        >
          <a
            href="/data"
          >
            Data
          </a>
        </li>
      </ul>
    </nav>
  </header>
</DocumentFragment>
`;
